src/main/java/com/ioet/bpm/peopletime/timeevents/boundaries/TimeEventController.java

```diff
@@ -38,6 +37,16 @@
         return new ResponseEntity<>(timeEvents, HttpStatus.OK);
     }
 
+    @ApiOperation(value = "Return the last active event belonging to one person", response = TimeEvent.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved last active event belonging to one person")
+    })
+    @GetMapping(path = "/last-active-event", produces = "application/json")
+    public ResponseEntity<TimeEvent> lastActiveTimeEventForOnePerson(@RequestParam(value = "personId") String personId) {
+        Optional<TimeEvent> lastActiveEvent = this.timeEventRepository.lastActiveTimeEvent(personId);
+        return new ResponseEntity<>(lastActiveEvent.get(), HttpStatus.OK);
+    }
```

Comment:
You should not create another method here. Instead, you should modify the method `getAllTimeEventsForOnePerson`. Probably something like this:

@ApiOperation(value = "Return a list of events belonging to one person", response = TimeEvent.class, responseContainer = "List")
    @ApiResponses(value = {
            @ApiResponse(code = 200, message = "Successfully retrieved all events belonging to one person")
    })
    @GetMapping(produces = "application/json")
    public ResponseEntity<?> findTimeEvents(@RequestParam(value = "personId") String personId, @RequestParam(value = "orderBy") String orderByCriteria, @RequestParam(value = "top") int top) {
               
        // If top is not provided you can return by default 10 items, otherwise top will define the quantity of numbers to return        
        if (orderByCriteria == null) {
            // Please create another ticket here to use top, we can also paginate the results but probably right now we don't need it
            Iterable<TimeEvent> timeEvents = this.timeEventRepository.findByPersonId(personId);
            return new ResponseEntity<>(timeEvents, HttpStatus.OK);
        } else if ("lastActive".equals(orderByCriteria)) {
            // Use the top here to return not only one but a certain number of items specified by top
            Optional<TimeEvent> lastActiveEvent = this.timeEventRepository.lastActiveTimeEvent(personId);
            return new ResponseEntity<>(lastActiveEvent.get(), HttpStatus.OK);
        } else {
            return new ResponseEntity<>("Provide orderBy criteria not supported", HttpStatus.BAD_REQUEST);
        }
        
        // A lot of if-else I suggest you to read a design pattern called Strategy to replace them
    }

Reply:
This code is put in a service.